### PR TITLE
FM-5711 add blacksmith for windows modules

### PR DESCRIPTION
**Pull request by tphoney**

### CODE REVIEW

File: moduleroot/Rakefile [hunk 1]
```diff
@@ -1,5 +1,6 @@
 require 'puppetlabs_spec_helper/rake_tasks'
 require 'puppet-lint/tasks/puppet-lint'
+require 'puppet_blacksmith/rake_tasks'
```
Comment (ferventcoder) [inline moduleroot/Rakefile]: This should probably be guarded for when the platform is not Windows.

File: config_defaults.yml [hunk 2]
```diff
@@ -23,6 +23,9 @@ Gemfile:
       - gem: puppetlabs_spec_helper
         version: '~>0.10.3'
       - gem: puppet_facts
+      # newer version required for ruby 1.9 compat
+      - gem: puppet-blacksmith
+        version: '>= 3.4.0'
```
Comment (ferventcoder) [inline config_defaults.yml]: Take a look at https://github.com/puppetlabs/puppetlabs-reboot/blob/00d4535df025b17d250f89229fd34fbb7f313361/.sync.yml#L13 for the additional bit for ensuring just on non-Windows platforms.

**Comment by ferventcoder** [comment]
@puppetlabs/modules @puppetlabs/windows anyone else for review? If not I will merge a little later.

**Comment by glennsarti** [comment]
Looks ok... would prefer to see output from a manual test that it does indeed do what it said it should.

**Comment by ferventcoder** [comment]
Yes, I need to reverse my changes first to be able to test this.

**Pull request closed without merging**